app/controllers/students_controller.rb

```diff
@@ -30,6 +29,11 @@ def show
     }
   end
 
+  def restricted_notes
+    raise Exceptions::EducatorNotAuthorized unless current_educator.can_view_restricted_notes
```

Comment:
To be safe this should probably also do `raise Exceptions::EducatorNotAuthorized unless current_educator.is_authorized_for_student(student)` to be safe, or get added to `authorize!` although I'm sure that this method is sufficient.